routers/web/repo/helper.go

```diff
@@ -21,3 +23,13 @@ func MakeSelfOnTop(ctx *context.Context, users []*user.User) []*user.User {
 	}
 	return users
 }
+
+func HandleGitError(ctx *context.Context, msg string, err error) {
+	if git.IsErrNotExist(err) {
+		ctx.Data["NotFoundPrompt"] = ctx.Locale.Tr("repo.tree_path_not_found", ctx.Repo.TreePath, ctx.Repo.BranchName)
+		ctx.Data["NotFoundGoBackURL"] = ctx.Repo.RepoLink + "/src/branch/" + url.PathEscape(ctx.Repo.BranchName)
```

Comment:
If you are visiting the tree at a given commit or tag, this should redirect to the root of the tree at that point.